traitsui/tests/test_ui.py

```diff
@@ -242,3 +250,200 @@ def test_invalid_state(self):
 
             obj.name = "too short"
             self.assertTrue(editor.invalid)
+
+
+# Regression test on an AttributeError commonly seen (enthought/traitsui#1145)
+# Code in ui_panel makes use toolkit specific attributes on the toolkit
+# specific Editor
+ToolkitSpecificEditor = toolkit_object("editor:Editor")
+
+
+class DummyObject(HasTraits):
+
+    number = Int()
+
+
+if is_qt():
+
+    from pyface.qt import QtGui, QtCore
+
+    class CustomWidget(QtGui.QWidget):
+
+        def __init__(self, editor, parent=None):
+            super(CustomWidget, self).__init__()
+            self._some_editor = editor
+
+        def sizeHint(self):
+            assert self._some_editor.factory is not None
+            return super().sizeHint()
```

Comment:
I see that when moving the call to `destroy_control` down, the test fails on this assertion, but where exactly is this `sizeHint` getting called?

Reply:
It's likely being called from Qt C++ code: as things are destroyed various widgets still in the layout are going to get asked how big they think they should be (which is what `sizeHint` does) and if the state is not in a good state then it will fail.

Reply:
Yep, what Corran's said. I should add better comments for this test code though... doing that now...